### PR TITLE
useCompletion refactoring.

### PR DESCRIPTION
**Pull request by lgrammel**
## Summary
- rename `callApi` to `callChatApi`
- introduce `callCompletionApi`
- refactor `useCompletion` implementations to use `callCompletionApi`

### CODE REVIEW

File: examples/nuxt-openai/server/api/completion.ts [hunk 1]
```diff
@@ -14,11 +14,7 @@ export default defineLazyEventHandler(async () => {
 
   return defineEventHandler(async (event: any) => {
     // Extract the `prompt` from the body of the request
-    const newVariable = await readBody(event);
-    console.log('nw', newVariable);
-    const { prompt } = newVariable as {
-      prompt: string;
-    };
+    const { prompt } = await readBody(event);
```
Comment (lgrammel) [inline examples/nuxt-openai/server/api/completion.ts:17]: Minor cleanup.

File: packages/core/shared/call-completion-api.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    const res = await fetch(api, {
+      method: 'POST',
+      body: JSON.stringify({
+        prompt,
+        ...body,
+      }),
+      credentials,
+      headers: {
+        'Content-Type': 'application/json',
```
Comment (lgrammel) [inline packages/core/shared/call-completion-api.ts:52]: Slight change: this is now the default for all frameworks.

File: packages/core/vue/use-completion.ts [hunk 3]
```diff
@@ -100,115 +99,36 @@ export function useCompletion({
...
+        error.value = err;
+      },
+      setAbortController: controller => {
+        abortController = controller;
+      },
+      onResponse,
+      onFinish,
+      onError,
+      onData: data => {
+        mutateStreamData(() => [...existingData, ...(data ?? [])]);
```
Comment (lgrammel) [inline packages/core/vue/use-completion.ts:129]: Switch from `streamData.value = ` to `mutateStreamData

**Review by MaxLeiter — APPROVED** [review]
🎉

**Pull request merged**